api/app/serializers/spree/v2/storefront/country_serializer.rb

```diff
@@ -12,6 +12,12 @@ class CountrySerializer < BaseSerializer
         end
 
         has_many :states, if: proc { |_record, params| params && params[:include_states] }
+
+        attribute :checkout_zone_applicable_states, if: proc { |_record, params| params && params[:current_store] } do |object, params|
```

Comment:
@nnande this cannot be an attribute, this should be a relation, same as line 14

Reply:
@damianlegawiec done via https://github.com/spree/spree/pull/10589/commits/106662c3f2d29d09439d379cfc8484419ad8df52